fastlane/lib/assets/completions/completion.fish

```diff
@@ -0,0 +1,37 @@
+# This function was taken from https://github.com/Carthage/Carthage/blob/master/Source/Scripts/carthage-fish-completion
+function __fish_prog_needs_subcommand
+  set cmd (commandline -opc)
+  if [ (count $cmd) -eq 1 -a $cmd[1] = 'fastlane' ]
+    return 0
+  end
+    return 1
+end
+
+if test -e "Fastfile"
+  set file "Fastfile"
+else if test -e "fastlane/Fastfile"
+  set file "fastlane/Fastfile"
+else if test -e ".fastlane/Fastfile"
+  set file ".fastlane/Fastfile"
+else
+  exit 1
+end
+
+set commands (string match --regex '.*lane\ \:(?!private_)([^\s]*)\ do' (cat $file))
```

Comment:
I agree that this could be improved. Let me know if you decide on any changes and I will create a new PR.

My branch is now up to date with fastlane's master :)